native/src/main/java/io/ballerina/lib/data/xmldata/utils/DataUtils.java

```diff
@@ -785,9 +826,17 @@ public static Type getTypeFromUnionType(Type childType, Object value) {
     }
 
     private static BMap<BString, Object> processParentAnnotation(Type type, BMap<BString, Object> record) {
+        BMap<BString, Object> annotations = ValueCreator.createMapValue();
+        if (type instanceof AnnotatableType annotatableType) {
+            annotations = annotatableType.getAnnotations();
+        }
+        Type referedType = TypeUtils.getReferredType(type);
+        if (type instanceof ReferenceType referenceType) {
+            annotations = mergeOriginalAndCurrentAnnotations(annotations, ((RecordType) referedType).getAnnotations());
```

Comment:
Why do we need to do this? Don't we already have annotations within the referred type? A referred type can contain another referred type. Are we expected to add annotations in that case as well?

Reply:
We needs to save the annotations of both referred type and the original types.
Normally this is not happen, this is a added benefit